eu_einvoice/hooks.py

```diff
@@ -143,7 +143,8 @@
 
 doc_events = {
 	"Sales Invoice": {
-		"validate": "eu_einvoice.european_e_invoice.custom.sales_invoice.validate_doc",
+		"before_submit": "eu_einvoice.european_e_invoice.custom.sales_invoice.validate_doc",
+		"before_save": "eu_einvoice.european_e_invoice.custom.sales_invoice.validate_doc",
```

Comment:
```suggestion
		"before_submit": "eu_einvoice.european_e_invoice.custom.sales_invoice.before_submit",
		"before_save": "eu_einvoice.european_e_invoice.custom.sales_invoice.before_save",
```
That is at least my personal preference, to have the run-methods extendable with further functions.
You would then call your validate_doc method in these run methods.

Example:
before_save(doc, event):
-> validate_doc(doc, event)
before_submit(doc, event):
-> validate_doc(doc, event)

Other thing: Best practice is also to sort the run methods sequentially.